src/rust/pipeline-ingress/tests/integration_test.rs

```diff
@@ -0,0 +1,137 @@
+#![cfg(feature = "new_integration_tests")]
+
+use std::time::{
+    Duration,
+    SystemTime,
+};
+
+use futures::StreamExt;
+use kafka::{
+    Consumer,
+    ConsumerError,
+};
+use opentelemetry::{
+    global,
+    sdk::propagation::TraceContextPropagator,
+};
+use rust_proto_new::graplinc::grapl::{
+    api::pipeline_ingress::v1beta1::{
+        client::{
+            HealthcheckClient,
+            PipelineIngressClient,
+        },
+        PublishRawLogRequest,
+    },
+    pipeline::{
+        v1beta1::RawLog,
+        v1beta2::Envelope,
+    },
+};
+use test_context::{
+    test_context,
+    AsyncTestContext,
+};
+use tracing_subscriber::{
+    prelude::*,
+    EnvFilter,
+};
+use uuid::Uuid;
+
+struct PipelineIngressTestContext {
+    grpc_client: PipelineIngressClient,
+}
+
+#[async_trait::async_trait]
+impl AsyncTestContext for PipelineIngressTestContext {
+    async fn setup() -> Self {
+        let (non_blocking, _guard) = tracing_appender::non_blocking(std::io::stdout());
+
+        // initialize json logging layer
+        let log_layer = tracing_subscriber::fmt::layer()
+            .json()
+            .with_writer(non_blocking);
+
+        // initialize tracing layer
+        global::set_text_map_propagator(TraceContextPropagator::new());
+        let tracer = opentelemetry_jaeger::new_pipeline()
+            .with_service_name("pipeline-ingress-integration-tests")
+            .install_batch(opentelemetry::runtime::Tokio)
+            .unwrap();
+
+        // register a subscriber
+        let filter = EnvFilter::from_default_env();
+        tracing_subscriber::registry()
+            .with(filter)
+            .with(log_layer)
+            .with(tracing_opentelemetry::layer().with_tracer(tracer))
+            .init();
+
```

Comment:
It should be at some point. The question is where to put it? Some type of overarching service abstraction would be the sensible place I think. I've been putting this sort of thing off until I have a concrete need (e.g. more than one service) to let the requirements drive the code design rather than my own "cleverness" or "creativity". I suspect it'll be something to do sooner rather than later though.